src/analysis/retail/deathknight/unholy/modules/spells/VileInfusion.tsx

```diff
@@ -0,0 +1,101 @@
+import { t } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, RemoveBuffEvent, FightEndEvent } from 'parser/core/Events';
+import { When } from 'parser/core/ParseResults';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+import { formatPercentage } from 'common/format';
+import { TIERS } from 'game/TIERS';
+
+class VileInfusion extends Analyzer {
+  lastBuffTime = 0;
+  totalBuffTime = 0;
+  buffIsUp = false;
+
+  constructor(options: Options) {
+    super(options);
+
+    this.active = this.selectedCombatant.has2PieceByTier(TIERS.T29);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.VILE_INFUSION_BUFF),
+      this.onBuffEvent,
+    );
+
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.VILE_INFUSION_BUFF),
+      this.onRemoveBuffEvent,
+    );
+
+    this.addEventListener(Events.fightend, this.onFightEnd);
+  }
+
+  onBuffEvent(event: ApplyBuffEvent) {
+    this.lastBuffTime = event.timestamp;
+    this.buffIsUp = true;
+  }
+
+  onRemoveBuffEvent(event: RemoveBuffEvent) {
+    this.totalBuffTime += event.timestamp - this.lastBuffTime;
+    this.buffIsUp = false;
+  }
+
+  onFightEnd(event: FightEndEvent) {
+    if (this.buffIsUp === true) {
+      this.totalBuffTime += event.timestamp - this.lastBuffTime;
+    }
+  }
+
+  suggestions(when: When) {
+    const averageBuffUptime = Number(this.totalBuffTime / this.owner.fightDuration);
```

Comment:
Same as mentioned on the gargoyle PR, stats used in multiple places should be placed in getters.